jerry-core/ecma/operations/ecma-bigint.c

```diff
@@ -76,56 +76,60 @@ ecma_bigint_parse_string (const lit_utf8_byte_t *string_p, /**< string represena
   ecma_bigint_digit_t radix = 10;
   uint32_t sign = (options & ECMA_BIGINT_PARSE_SET_NEGATIVE) ? ECMA_BIGINT_SIGN : 0;
 
-  if (size >= 3 && string_p[0] == LIT_CHAR_0)
+  const lit_utf8_byte_t *nonws_start_p = ecma_string_trim_front (string_p, string_p + size);
+  lit_utf8_size_t nonws_size = (lit_utf8_size_t) ((string_p + size) - nonws_start_p);
+  size = nonws_size;
+
+  if (size >= 3 && nonws_start_p[0] == LIT_CHAR_0)
   {
-    if (string_p[1] == LIT_CHAR_LOWERCASE_X || string_p[1] == LIT_CHAR_UPPERCASE_X)
+    if (nonws_start_p[1] == LIT_CHAR_LOWERCASE_X || nonws_start_p[1] == LIT_CHAR_UPPERCASE_X)
     {
       radix = 16;
-      string_p += 2;
+      nonws_start_p += 2;
       size -= 2;
     }
-    else if (string_p[1] == LIT_CHAR_LOWERCASE_O || string_p[1] == LIT_CHAR_UPPERCASE_O)
+    else if (nonws_start_p[1] == LIT_CHAR_LOWERCASE_O || nonws_start_p[1] == LIT_CHAR_UPPERCASE_O)
     {
       radix = 8;
-      string_p += 2;
+      nonws_start_p += 2;
       size -= 2;
     }
-    else if (string_p[1] == LIT_CHAR_LOWERCASE_B || string_p[1] == LIT_CHAR_UPPERCASE_B)
+    else if (nonws_start_p[1] == LIT_CHAR_LOWERCASE_B || nonws_start_p[1] == LIT_CHAR_UPPERCASE_B)
     {
       radix = 2;
-      string_p += 2;
+      nonws_start_p += 2;
       size -= 2;
     }
   }
   else if (size >= 2)
   {
-    if (string_p[0] == LIT_CHAR_PLUS)
+    if (nonws_start_p[0] == LIT_CHAR_PLUS)
     {
       size--;
-      string_p++;
+      nonws_start_p++;
     }
-    else if (string_p[0] == LIT_CHAR_MINUS)
+    else if (nonws_start_p[0] == LIT_CHAR_MINUS)
     {
       sign = ECMA_BIGINT_SIGN;
       size--;
-      string_p++;
+      nonws_start_p++;
     }
   }
   else if (size == 0)
   {
     return ECMA_BIGINT_ZERO;
   }
 
-  const lit_utf8_byte_t *string_end_p = string_p + size;
+  const lit_utf8_byte_t *string_end_p = nonws_start_p + size;
```

Comment:
I think we don't need to introduce new variables. Just move this defintion to the beginning of the function.

```
const lit_utf8_byte_t *string_end_p = string_p + size;
string_p = ecma_string_trim_front (string_p, string_end_p);
size = (lit_utf8_size_t ) (string_end_p - string_p);
```